tests/php/test-class-amp-gtag-script-sanitizer.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Class AMP_GTag_Script_Sanitizer_Test.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+use AmpProject\AmpWP\ValidationExemption;
+use AmpProject\Dom\Document;
+use AmpProject\Html\Tag;
+use AmpProject\AmpWP\Option;
+use AmpProject\AmpWP\Tests\TestCase;
+
+/**
+ * Tests the GTag Script Sanitizer class.
+ *
+ * @coversDefaultClass AMP_GTag_Script_Sanitizer
+ */
+class AMP_GTag_Script_Sanitizer_Test extends TestCase {
+
+	/**
+	 * Sandboxing enabled.
+	 *
+	 * @var bool
+	 */
+	private $sandboxing_enabled;
+
+	/**
+	 * Sandboxing level.
+	 *
+	 * @var int
+	 */
+	private $sandboxing_level;
+
+	/**
+	 * XPath query to find the scripts element.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_XPATH = '//script[ ( @async and starts-with( @src, "https://www.googletagmanager.com/gtag/js" ) ) or contains( text(), "function gtag(" ) ]';
+
+	/**
+	 * HTML markup with gtag script.
+	 *
+	 * @var string
+	 */
+	const GTAG_HTML_MARKUP = '<html><head></head><body><script async src="https://www.googletagmanager.com/gtag/js?id=xxxxxx"></script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","xxxxxx")</script><script src="https://example.com/pixel/js"></script></body></html>';
+
+	/**
+	 * Enable sandboxing for the test.
+	 *
+	 * @param bool $enabled Whether sandboxing is enabled.
+	 * @param int $sandboxing_level The sandboxing level to set.
+	 */
+	public function enable_sandboxing( $enable = true, $level = 3 ) {
+		AMP_Options_Manager::update_option( Option::SANDBOXING_ENABLED, $enable );
+		if ( $enable ) {
+			AMP_Options_Manager::update_option( Option::SANDBOXING_LEVEL, $level );
+		}
+	}
```

Comment:
This method is now unused and can be removed, right?

```suggestion
```

Reply:
Ahh. yes. Thanks.